GitUI/CommandsDialogs/RepoHosting/ViewPullRequestsForm.cs

```diff
@@ -395,16 +401,19 @@ private void _addAsRemoteAndFetch_Click(object sender, EventArgs e)
             try
             {
                 var remoteName = _currentPullRequestInfo.Owner;
-                var remoteUrl = _currentPullRequestInfo.HeadRepo.CloneReadOnlyUrl;
+                var remoteUrl = _cloneWithHttps ? _currentPullRequestInfo.HeadRepo.CloneReadWriteUrl : _currentPullRequestInfo.HeadRepo.CloneReadOnlyUrl;
                 var remoteRef = _currentPullRequestInfo.HeadRef;
 
                 var existingRepo = _hostedRemotes.FirstOrDefault(el => el.Name == remoteName);
                 if (existingRepo != null)
                 {
-                    if (existingRepo.GetHostedRepository().CloneReadOnlyUrl != remoteUrl)
+                    var hostedRepository = existingRepo.GetHostedRepository();
+                    hostedRepository.CloneProtocol = _cloneWithHttps ? GitProtocol.Https : GitProtocol.Ssh;
+                    var hostedRepositoryCloneUrl = _cloneWithHttps ? hostedRepository.CloneReadWriteUrl : hostedRepository.CloneReadOnlyUrl;
```

Comment:
I suggest to factor out setting `CloneProtocol` and `*CloneUrl` and
to comment why the `CloneReadWriteUrl` is the right choice for HTTPS.

Reply:
I don't like the solution I adopted to fix the problem but that was due to the existing code https://github.com/gitextensions/gitextensions/blob/553db71b4487d38778b40e6d9d3fe43ed2a0a246/Plugins/GitHub3/GitHubRepo.cs#L73 that do a bad abstraction (in this case) of the underlying code (that would be better in my case):
and https://github.com/gitextensions/Git.hub/blob/master/Git.hub/Repository.cs#L39

We have to set the the protocol https to get the good ReadWrite clone url...

I don't understand what you saying, so perhaps you could better explain (for a not native english speaker) or write some (pseudo) code?

Reply:
My point is "DRY!". You do two things twice the same way.

`private bool _cloneWithHttps` should be replaced with `private GitProtocol _cloneProtocol`.

The selection of the `cloneUrl` should be factored out, e.g. as a property, and to be explained:
```C#
        /// <summary>
        /// explain why HTTPS must be handled differently
        /// </summary>
        private string CloneUrl { get => _cloneProtocol == GitProtocol.Https ? _currentPullRequestInfo.HeadRepo.CloneReadWriteUrl : _currentPullRequestInfo.HeadRepo.CloneReadOnlyUrl; }
```